benchmark/torch/DT/data_loader.py

```diff
@@ -0,0 +1,114 @@
+            returns.append(path['rewards'].sum())
+        traj_lens, returns = np.array(traj_lens), np.array(returns)
+
+        # used for input normalization
+        states = np.concatenate(states, axis=0)
+        self.state_mean, self.state_std = np.mean(states, axis=0), np.std(states, axis=0) + 1e-6
+
+        num_timesteps = sum(traj_lens)
+
+        # only train on top pct_traj trajectories (for %BC experiment)
+        num_timesteps = max(int(pct_traj * num_timesteps), 1)
+        sorted_inds = np.argsort(returns)  # lowest to highest
+        num_trajectories = 1
+        timesteps = traj_lens[sorted_inds[-1]]
+        ind = len(trajectories) - 2
+        while ind >= 0 and timesteps + traj_lens[sorted_inds[ind]] <= num_timesteps:
+            timesteps += traj_lens[sorted_inds[ind]]
+            num_trajectories += 1
+            ind -= 1
+        sorted_inds = sorted_inds[-num_trajectories:]
+
+        # used to reweight sampling so we sample according to timesteps instead of trajectories
+        p_sample = traj_lens[sorted_inds] / sum(traj_lens[sorted_inds])
+
+        logger.info(f'{len(traj_lens)} trajectories, {num_timesteps} timesteps found')
+        logger.info(f'Average return: {np.mean(returns):.2f}, std: {np.std(returns):.2f}')
+        logger.info(f'Max return: {np.max(returns):.2f}, min: {np.min(returns):.2f}')
+        self.trajectories = trajectories
+        self.num_trajectories = num_trajectories
+        self.p_sample = p_sample
+        self.sorted_inds = sorted_inds
+        self.state_dim = trajectories[0]['observations'].shape[1]
+        self.act_dim = trajectories[0]['actions'].shape[1]
+
+    def get_batch(self, batch_size=256, max_len=20):
+        batch_inds = np.random.choice(
+            np.arange(self.num_trajectories),
+            size=batch_size,
+            replace=True,
+            p=self.p_sample,  # reweights so we sample according to timesteps
+        )
+
+        s, a, r, d, rtg, timesteps, mask = [], [], [], [], [], [], []
+        for i in range(batch_size):
+            traj = self.trajectories[int(self.sorted_inds[batch_inds[i]])]
+            si = random.randint(0, traj['rewards'].shape[0] - 1)
+
+            # get sequences from dataset
+            s.append(traj['observations'][si:si + max_len].reshape(1, -1, self.state_dim))
+            a.append(traj['actions'][si:si + max_len].reshape(1, -1, self.act_dim))
+            r.append(traj['rewards'][si:si + max_len].reshape(1, -1, 1))
+            if 'terminals' in traj:
+                d.append(traj['terminals'][si:si + max_len].reshape(1, -1))
+            else:
+                d.append(traj['dones'][si:si + max_len].reshape(1, -1))
+            timesteps.append(np.arange(si, si + s[-1].shape[1]).reshape(1, -1))
+            timesteps[-1][timesteps[-1] >= self.max_ep_len] = self.max_ep_len - 1  # padding cutoff
+            rtg.append(self.discount_cumsum(traj['rewards'][si:], gamma=1.)[:s[-1].shape[1] + 1].reshape(1, -1, 1))
```

Comment:
As mentioned above, if changed above, then change `gamma=1` to `gamma=gamma`

Reply:
The response to this comment can be found above.